src/core/audio/AudioPortAudio.cpp

```diff
@@ -248,6 +248,21 @@ void gui::AudioPortAudioSetupWidget::show()
 	AudioDeviceSetupWidget::show();
 }
 
+PortAudioInitializationGuard::PortAudioInitializationGuard()
+	: m_error(Pa_Initialize())
+{
+}
+
+PortAudioInitializationGuard::~PortAudioInitializationGuard()
+{
+	if (m_error == paNoError)
+	{
+		Pa_Terminate();
+		return;
+	}
+
+	std::cerr << "Failed to initialize PortAudio: " << Pa_GetErrorText(m_error);
```

Comment:
I think it's better to print the error message in the constructor, not in the destructor.

Reply:
Right, fixed in 94dc9cd916fc59f2179400685dbc3198c0022123.